### DISCUSSION TITLE
Can not set job data map with oracle

### DISCUSSION THREAD

**Discussion by db-conception** [quartznet/quartznet#2264] [Q&A] [answered]
**Describe the bug**

With Oracle, we can not save datamap as blob. Quartz send a BLOB with "-1" as size. But Oracle throw an out of range exception
![image (2)](https://github.com/quartznet/quartznet/assets/43586669/586ddee2-6199-4890-8eb7-cbce410ca5dc)
![image (3)](https://github.com/quartznet/quartznet/assets/43586669/b02cd214-4d15-469d-93d3-1d7cf9e8fa1c)

**Version used**

I am using quartz v3.8.0

**To Reproduce**

Create a quartz.net implementation with Oracle packages: 
```xml
    <PackageReference Include="Oracle.EntityFrameworkCore" Version="7.21.13" />
    <PackageReference Include="Oracle.ManagedDataAccess.Core" Version="3.21.130" />
```

In the config we have 
```json
"Quartz": {
  "quartz.jobStore.type": "Quartz.Impl.AdoJobStore.JobStoreTX, Quartz",
  "quartz.jobStore.driverDelegateType": "Quartz.Impl.AdoJobStore.SqlServerDelegate, Quartz",
  "quartz.jobStore.tablePrefix": "QRTZ_",
  "quartz.jobStore.dataSource": "default",
  "quartz.dataSource.default.connectionString": "User Id=scheduler;Password=******;Data Source=localhost:1521/FREEPDB1",
  //"quartz.dataSource.default.connectionString": "Server=localhost;Database=Scheduler;User Id=sa;Pwd=********;Trusted_Connection=false;MultipleActiveResultSets=true;Encrypt=False",
  "quartz.dataSource.default.provider": "OracleODPManaged",
  "quartz.jobStore.useProperties": "false",
  "quartz.serializer.type": "json",
  "quartz.scheduler.instanceName": "QuartzServiceScheduler"
```

**Expected behavior**

I expect to save my datamap without problem

**Additional context**
C# 7

**Comment by lahma** [comment, marked as answer]
You are using Microsoft SQL Server delegate. Try using `OracleDelegate`.

**Comment by db-conception** [comment]
It solves my problem problem thank you very much !

Did I miss a documentation about how to connect to Oracle database ?

**Reply by lahma** [reply]
* https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/job-stores.html#configuring-quartz-to-use-jobstoretx
* https://www.quartz-scheduler.net/documentation/quartz-3.x/configuration/reference.html#quartz-jobstore-driverdelegatetype
